core/src/hir/type_context.rs

```diff
@@ -144,10 +167,21 @@ impl TypeContext {
         self.enums.index(id.0)
     }
 
+    pub fn resolve_trait(&self, id: TraitId) -> &TraitDef {
+        self.traits.index(id.0)
+    }
+
     /// Resolve and format a named type for use in diagnostics
     /// (don't apply rename rules and such)
     pub fn fmt_type_name_diagnostics(&self, id: TypeId) -> Cow<str> {
-        self.resolve_type(id).name().as_str().into()
+        self.fmt_symbol_name_diagnostics(id.into())
```

Comment:
nit: i'd have fmt_symbol_name call this one, not the other way around

(feels pointless to convert to a SYmbolId and immediately match on it)

Reply:
good point